### PR TITLE
Fix potential null reference exceptions in ActivityHelper 

### LINKED ISSUE #1282: Null Pointer Exception when using bytecode instrumentation
# Bug Report

## Symptom

**Describe the bug**

I am implementing our own bytecode instrumentation and sometimes hit NPE:

```
[2022-09-27T13:52:14.9181394Z] [Error] Object reference not set to an instance of an object. 
Exception: Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object.
   at OpenTelemetry.AutoInstrumentation.Util.ActivityHelper.StartActivityWithTags(ActivitySource activitySource, String operationName, ActivityKind kind, ITags tags)
   at OpenTelemetry.AutoInstrumentation.Instrumentations.Odp.OdpCommon.OnMethodBeginConnection[TTarget](TTarget instance, String spanNamePrefix)
   at OdpOpenConnectionIntegration.OnMethodBegin(OracleConnection )
   at OpenTelemetry.AutoInstrumentation.CallTarget.Handlers.BeginMethodHandler`2.Invoke(TTarget instance)
   at OpenTelemetry.AutoInstrumentation.CallTarget.CallTargetInvoker.BeginMethod[TIntegration,TTarget](TTarget instance)
   at Oracle.ManagedDataAccess.Client.OracleConnection.Open()
```

Looked into source code and I found [this line](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/de45afdb36733468fa2e78a7ba7197d54abc0c73/src/OpenTelemetry.AutoInstrumentation/Util/ActivityHelper.cs#L42) . It is possible that activitySource.StartActivity() returns null. So we'd better check for it in [the following logic](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/de45afdb36733468fa2e78a7ba7197d54abc0c73/src/OpenTelemetry.AutoInstrumentation/Util/ActivityHelper.cs#L49).

**Runtime environment:**

- OpenTelemetry Automatic Instrumentation version: 0.3.1
- OS: Windows 10
- .NET version: .NET Framework 4.6.2

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #1282

## What

ActivityHelper - fix potential null reference exceptions.
Cover class by tests

## Tests

CI + new test in PR.

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- ~~[ ] Documentation is updated.~~
- [x] ~~New~~ features are covered by tests.

### CODE REVIEW

**Comment by Kielek** [comment]
@muhaook FYI

**Review by pellared — APPROVED** [review]

File: test/OpenTelemetry.AutoInstrumentation.Tests/Util/ActivityHelperTests.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    [Theory]
+    [InlineData(ActivityKind.Internal)]
+    [InlineData(ActivityKind.Server)]
+    [InlineData(ActivityKind.Client)]
+    [InlineData(ActivityKind.Producer)]
+    [InlineData(ActivityKind.Consumer)]
+    public void StartActivityWithTags_ReturnsActivity_WhenThereIsActivityListener(ActivityKind kind)
+    {
+        var activitySource = new ActivitySource("test-source");
```
Comment (RassK) [inline test/OpenTelemetry.AutoInstrumentation.Tests/Util/ActivityHelperTests.cs:100]: might be safer to add `using` here when global registration is used?

**Comment by muhaook** [comment]
Looks good to me[👍](https://whatemoji.org/blog/thumbs-up-emoji-guide/#:~:text=The%20%F0%9F%91%8D%20Thumbs%20Up%20emoji%20is%20optimistic.%20Everything,that%20you%20like%20something%20and%20approve%20of%20it.) Thanks

**Review by rajkumar-rangaraj — APPROVED** [review]

File: test/OpenTelemetry.AutoInstrumentation.Tests/Util/ActivityHelperTests.cs (hunk 1 above)
Comment (Kielek) [reply test/OpenTelemetry.AutoInstrumentation.Tests/Util/ActivityHelperTests.cs:100]: I have missed that ActivitySource implements IDisposable.

I will fix it tommorow. Please do not merge until I fix it.

**Review by pjanotti — CHANGES_REQUESTED** [review]
Blocking just to avoid accidental merges per @Kielek request.

**Review by zacharycmontoya — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation/Util/ActivityHelper.cs [hunk 2]
```diff
@@ -30,44 +30,30 @@ internal static class ActivityHelper
     /// <param name="exception">The exception.</param>
     public static void SetException(this Activity activity, Exception exception)
     {
-        if (exception != null)
+        if (activity == null || exception == null)
```
[the remark this answers is not in the document]
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation/Util/ActivityHelper.cs]: 3a3f61e3ca5f5b55a68c3745833cd2acb1335361

File: src/OpenTelemetry.AutoInstrumentation/Util/ActivityHelper.cs [hunk 3]
```diff
@@ -16,58 +16,53 @@
...
         }
+
+        activity.SetStatus(Status.Error.WithDescription(exception.Message));
+        activity.RecordException(exception);
     }
 
     public static Activity StartActivityWithTags(this ActivitySource activitySource, string operationName, ActivityKind kind, ITags tags)
     {
-        var activity = activitySource.StartActivity(operationName, kind);
+        var activity = activitySource?.StartActivity(operationName, kind);
```
Comment (pellared) [inline src/OpenTelemetry.AutoInstrumentation/Util/ActivityHelper.cs:53]: Do we want to log if `activitySource` is `null` in the same way as it is done in `SetException` extension method?

**Review by pellared — APPROVED** [review]

**Pull request merged**